pkg/client/typed/scaler/scaler.go

```diff
@@ -176,36 +217,41 @@ type extScale struct {
 	scale
 }
 
-func (s *extScale) Selector() (string, error) {
+func (s *extScale) Selector() (labels.Selector, error) {
 	seli, err := field(s.obj.Object, "status", "selector")
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	selMap, ok := seli.(map[string]interface{})
 	if !ok {
-		return "", fmt.Errorf("`Selector` isn't of expected type")
+		return nil, fmt.Errorf("`Selector` isn't of expected type")
 	}
 
 	selector := make(map[string]string)
 	for key, val := range selMap {
 		selector[key] = val.(string)
 	}
 
-	return labels.SelectorFromSet(selector).String(), nil
+	return labels.SelectorFromSet(selector), nil
 }
 
 type asScale struct {
 	scale
 }
 
-func (s *asScale) Selector() (string, error) {
+func (s *asScale) Selector() (labels.Selector, error) {
 	seli, err := field(s.obj.Object, "status", "selector")
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	selector, ok := seli.(string)
 	if !ok {
-		return "", fmt.Errorf("`Selector` isn't of expected type")
+		return nil, fmt.Errorf("`Selector` isn't of expected type")
+	}
+	// TODO: Eliminate deserialization/reserialization of the selector.
```

Comment:
I think implementing this TODO is a requirement-- clients need to just pass-through selectors they don't understand. It shouldn't be too hard to add a pass-through implementation in the labels package, I think?